test/core-test.el

```diff
@@ -261,6 +261,23 @@ The prompt is chosen according to `purpose-preferred-prompt'."
   (purpose-insert-user-input "foo")
   (should (equal (purpose-read-purpose "Purpose: " '(foo bar baz)) 'foo)))
 
+(ert-deftest purpose-test-default-purpose-when-visiting-file ()
+  "Test that the default purpose for a buffer visiting a file is 'edit.
+Also test that if there was a predefined purpose for that buffer
+it gets that one, and that the default purpose for a buffer not
+visiting a file is still `default-purpose'."
+  (purpose-with-temp-config
+   nil '(("foo" . bar)) nil
+   (find-file "foo")
+   (should (equal (purpose-buffer-purpose (get-buffer "foo")) 'bar))
+   (kill-buffer "foo")
+   (find-file "baz")
+   (should (equal (purpose-buffer-purpose (get-buffer "baz")) 'edit))
+   (kill-buffer "baz")
+   (get-buffer-create "yolo")
+   (should (equal (purpose-buffer-purpose (get-buffer "yolo")) default-purpose))
+   (kill-buffer "yolo")))
+
```

Comment:
Use `unwind-protect` and `purpose-kill-buffers-safely`. This ensures proper tear-down of the test, and separate the tear-down from what's actually being tested. See other tests in the same file for examples.